src/main/java/com/hibernate/service/impl/AuthenticationServiceImpl.java

```diff
@@ -4,20 +4,24 @@
 import com.hibernate.lib.Service;
 import com.hibernate.model.User;
 import com.hibernate.service.AuthenticationService;
+import com.hibernate.service.ShoppingCardService;
 import com.hibernate.service.UserService;
 import com.hibernate.util.HashUtil;
 import java.util.Optional;
 
 @Service
 public class AuthenticationServiceImpl implements AuthenticationService {
     private UserService userService = new UserServiceImpl();
+    private ShoppingCardService shoppingCardService = new ShoppingCardServiceImpl();
```

Comment:
```suggestion
    private UserService userService = new UserServiceImpl();
    private ShoppingCardService shoppingCardService = new ShoppingCardServiceImpl();
```
we should use @Inject here